controller_manager/test/test_load_controller.cpp

```diff
@@ -183,192 +213,314 @@ TEST_F(TestLoadController, configure_controller)
-        start_controllers, stop_controllers,
-        STRICT, true, rclcpp::Duration(0, 0))
-    ) << "STRICT switch with nonexistent controller specified";
-  }
+// TODO(anyone): For Galactic+, follow up on https://github.com/ros-controls/ros2_control/issues/406
+#ifdef __APPLE__
+INSTANTIATE_TEST_CASE_P(
+  EmptyListOrNonExistentTest,
+  SwitchTest,
+  ::testing::Values(
+    // empty lists
+    std::make_tuple(
+      controller_interface::return_type::OK, UNSPECIFIED, EMPTY_STR_VEC,
+      EMPTY_STR_VEC, "Switch with no controllers specified"),
+    std::make_tuple(
+      controller_interface::return_type::OK, STRICT, EMPTY_STR_VEC, EMPTY_STR_VEC,
+      "Switch with no controllers specified"),
+    std::make_tuple(
+      controller_interface::return_type::OK, BEST_EFFORT, EMPTY_STR_VEC,
+      EMPTY_STR_VEC, "Switch with no controllers specified"),
+    // combination of empty and non-existent controller
+    std::make_tuple(
+      controller_interface::return_type::OK, UNSPECIFIED, NONEXISTENT_CONTROLLER,
+      EMPTY_STR_VEC, "Switch with nonexistent controller specified"),
+    std::make_tuple(
+      controller_interface::return_type::ERROR, STRICT, NONEXISTENT_CONTROLLER,
+      EMPTY_STR_VEC, "Switch with nonexistent start controller specified"),
+    std::make_tuple(
+      controller_interface::return_type::OK, BEST_EFFORT, NONEXISTENT_CONTROLLER,
+      EMPTY_STR_VEC, "Switch with nonexistent start controller specified"),
+    std::make_tuple(
+      controller_interface::return_type::OK, UNSPECIFIED, EMPTY_STR_VEC,
+      NONEXISTENT_CONTROLLER, "Switch with nonexistent stop controller specified"),
+    std::make_tuple(
+      controller_interface::return_type::ERROR, STRICT, EMPTY_STR_VEC,
+      NONEXISTENT_CONTROLLER, "Switch with nonexistent stop controller specified"),
+    std::make_tuple(
+      controller_interface::return_type::OK, BEST_EFFORT, EMPTY_STR_VEC,
+      NONEXISTENT_CONTROLLER, "Switch with nonexistent stop controller specified"),
+    std::make_tuple(
+      controller_interface::return_type::OK, UNSPECIFIED, NONEXISTENT_CONTROLLER,
+      NONEXISTENT_CONTROLLER, "Switch with nonexistent start and stop controllers specified"),
+    std::make_tuple(
+      controller_interface::return_type::ERROR, STRICT, NONEXISTENT_CONTROLLER,
+      NONEXISTENT_CONTROLLER, "Switch with nonexistent start and stop controllers specified"),
+    std::make_tuple(
+      controller_interface::return_type::OK, BEST_EFFORT, NONEXISTENT_CONTROLLER,
+      NONEXISTENT_CONTROLLER, "Switch with nonexistent start and stop controllers specified"),
+    // valid controller used
+    std::make_tuple(
+      controller_interface::return_type::ERROR, STRICT, NONEXISTENT_CONTROLLER,
+      VALID_CONTROLLER, "Switch with valid stopped controller specified"),
+    std::make_tuple(
+      controller_interface::return_type::OK, BEST_EFFORT, NONEXISTENT_CONTROLLER,
+      VALID_CONTROLLER, "Switch with valid stopped controller specified"),
```

Comment:
Error messages do not seem to be right, at least they are not clear. What about to make them longer and more descriptive?